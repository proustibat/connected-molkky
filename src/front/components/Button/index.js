import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ onClick, children, disabled = false }) => (
  <button
    type="button"
    className={`waves-effect waves-light btn btn-large ${disabled ? 'disabled' : ''}`}
    onClick={onClick}
  >
    {children}
  </button>
);

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  disabled: PropTypes.bool.isRequired,
};

export default Button;
