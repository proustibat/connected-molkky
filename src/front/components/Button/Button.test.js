import Button from './index';
import React from 'react';
import { shallow } from 'enzyme';

describe('Button', () => {
  it('should render the component correctly when it is enabled', () => {
    // Given / When
    const component = shallow(<Button onClick={() => {}}>click me</Button>);

    // Then
    expect(component).toMatchSnapshot();
  });

  it('should render the component correctly when it is disabled', () => {
    // Given / When
    const component = shallow(<Button onClick={() => {}} disabled>click me</Button>);

    // Then
    expect(component).toMatchSnapshot();
  });

  it('should inherit style', () => {
    // Given / When
    const style = { color: 'blue', width: 'auto', margin: 0 };
    const component = shallow(<Button onClick={() => {}} style={style}>click me</Button>);

    // Then
    expect(component.find('button').props().style).toMatchObject(style);
  });

  it('should handle click', () => {
    // Given / When
    const onClick = jest.fn();
    const component = shallow(<Button onClick={onClick}>click me</Button>);

    // When
    component.simulate('click');

    // Then
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
