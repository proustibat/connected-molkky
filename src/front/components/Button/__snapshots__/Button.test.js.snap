// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the component correctly when it is disabled 1`] = `
<button
  className="waves-effect waves-light z-depth-1 btn btn-large disabled"
  onClick={[Function]}
  style={Object {}}
  type="button"
>
  click me
</button>
`;

exports[`Button should render the component correctly when it is enabled 1`] = `
<button
  className="waves-effect waves-light z-depth-1 btn btn-large"
  onClick={[Function]}
  style={Object {}}
  type="button"
>
  click me
</button>
`;
